pkg/controllers/provisioning/v1alpha1/reallocation/terminate.go

```diff
@@ -60,18 +61,16 @@ func (t *Terminator) cordonNodes(ctx context.Context, provisioner *v1alpha1.Prov
 	}
 	// 2. Cordon nodes
 	for _, node := range nodeList {
-		if !node.Spec.Unschedulable {
-			persisted := node.DeepCopy()
-			node.Spec.Unschedulable = true
-			node.Labels = functional.UnionStringMaps(
-				node.Labels,
-				map[string]string{v1alpha1.ProvisionerPhaseLabel: v1alpha1.ProvisionerDrainingPhase},
-			)
-			if err := t.kubeClient.Patch(ctx, node, client.MergeFrom(persisted)); err != nil {
-				return fmt.Errorf("patching node %s, %w", node.Name, err)
-			}
-			zap.S().Debugf("Cordoned node %s", node.Name)
+		persisted := node.DeepCopy()
+		node.Spec.Unschedulable = true
+		node.Labels = functional.UnionStringMaps(
+			node.Labels,
+			map[string]string{v1alpha1.ProvisionerPhaseLabel: v1alpha1.ProvisionerDrainingPhase},
+		)
+		if err := t.kubeClient.Patch(ctx, node, client.MergeFrom(persisted)); err != nil {
```

Comment:
Is it possible that this sends unnecessary requests? Should we check if the patch is empty or not?

Reply:
I don't think that's necessary, for the same reason as above. After a node is labeled with the `v1alpha1.ProvisionerPhaseLabel` it won't be cordoned again. The cordon should only be logged once per node that is cordoned.

Reply:
Ah you're right, since we change the phase to draining, we don't need to filter. At one point, we were discussing collapsing Draining and Terminable into a single label value, so this would've been necessary.